### PR TITLE
Readds irradiate failure to nuclear gun on EMP

### PR DESCRIPTION
**Pull request by Jordie0608**
Readds unique failure mechanic to advanced energy gun that was removed in  #17576 along with reliability. If the gun gets EMPed, each time the gun is fired there's a low chance it'll irradiate you, if this happens enough it'll stop auto-charging. Further EMPs the gun increases this chance. This also fixes part of the icon being missing.

I considered adding additonal boons to the gun however RD-made auto-charging seems reasonable enough for the rare occurance of irradiating you. Thoughts?

:cl:
add: Advanced Energy Guns once again have a chance to irradiate you if EMPed.
/:cl:

### CODE REVIEW

**Comment by PKPenguin321** [comment]
Make it explode

**Comment by Jordie0608** [comment]
It explodes with radiation.

**Comment by kevinz000** [comment]
This seems too hard of a nerf because someone can't deal with getting dunked.
Also, why not make it only have a fail chance when it gets emped? No one is going to use a radioactive gun that will do it from the start.

**Comment by YakumoChen** [comment]
could we please not have reliability back in along with all the bugs it brought? The reliability thing is stupid anyway since you can make it normal again just by deconstructing a half a dozen guns.

**Comment by Iamgoofball** [comment]
@YakumoChen This doesn't use reliability.

**Comment by XDTM** [comment]
Yeah, i doubt it needs a nerf, make the base chance of radiation 0 and have it increase with each EMP

**Comment by GunHog** [comment]
Or a POWERFUL radiation blast without all this fail chance stuff. None of the Sec players to which I offer this gun would take it in the past without me confirming that it is safe. Who would blame them?

Should we even be indulging the salt from the original PR? If I remember correctly, as I was the AI that nuke round, PKP was probably the Op spamming EMPs and is angry about having failed to completely neutralize the AE gun user. 

EMPs breaking the gun and dropping heavy rads is fine, I agree that EMPs should ruin the gun. But it should not gib the user (original PR effect), and should not be rendered undesirable in normal use (this PR). Sec players are not going to want this thing if this PR is merged.

**Comment by Jordie0608** [comment]
I've no affiliation with whatever happen to PKP, his PR merely alerted me that the original behaviour of the gun, as per this PR, had been removed. I made this to restore it since I'm totally not biased towards the animated fail state sprites I spent 10 minutes making.

It's my opinion the gun shouldn't be entirely danger-less as I find that to be kinda boring and, dare i say, hugboxy. But you know, ~opinions~. EMP-only failure sounds fine however.

**Comment by GunHog** [comment]
>It's my opinion the gun shouldn't be entirely danger-less as I find that to be kinda boring and, dare i say, hugboxy. But you know, ~opinions~. EMP-only failure sounds fine however.

Yes, make it WRECK the user with a high rad dose, but it should be survivable. Rads poison, mutate, and stun you for a loooong time. Combine that with zeroing out the gun's charge, and you have your EMP response.

**Comment by Durkel** [comment]
You could get the reliability of the old gun up to 100% to avoid the radiation blasts, anyone with a sane mind would do that. If you're going to add some shitty mechanic back in to gimp the gun then just remove it and add a new unique weapon that isn't just a recharging HoS laser.

**Comment by calzilla1** [comment]
Make it so that you can screw driver it for MAX POWER mode which makes it charge 5 times faster but moderately irradiate you

**Comment by Supermichael777** [comment]
no let you screwdriver it to remove the charging limiter. twice as deep a battery but is this thing getting hot..? (Muffled boom)

**Comment by PKPenguin321** [comment]
it should still explode

**Comment by kevinz000** [comment]
Please no, R&D already had its sub memechine guns taken, don't ruin AEGs again ;_;

**Comment by PKPenguin321** [comment]
yes poor poor R&D it is the least respected department it is so weak and doesn't have a single good thing going for it

**Comment by YakumoChen** [comment]
you ever find it weird that RnD weapons are now no longer straight upgrades but now they're all weird and impractical gimmick weapons instead?

Xray is a skirting exception, but only because it could hit random innocent people down the line

**Comment by PKPenguin321** [comment]
As they should be

**Comment by Supermichael777** [comment]
>Xray is a skirting exception, but only because it could hit random innocent people down the line

your acting like unintentional chaos is a bad thing.

File: code/modules/projectiles/guns/energy/nuclear.dm [hunk 1]
```diff
@@ -79,3 +79,46 @@
...
+				M << "<span class='userdanger'>Your [name]'s reactor overloads!</span>"
+
+/obj/item/weapon/gun/energy/gun/nuclear/emp_act(severity)
+	..()
+	fail_chance = max(fail_chance - round(15/severity), 0)
+
+/obj/item/weapon/gun/energy/gun/nuclear/update_icon()
+	..()
+	if(crit_fail)
+		overlays += "[icon_state]_fail_3"
```
Comment (optimumtact) [inline code/modules/projectiles/guns/energy/nuclear.dm]: add_overlay()

Comment (optimumtact) [inline code/modules/projectiles/guns/energy/nuclear.dm]: @Jordie0608 please

**Comment by GunHog** [comment]
Now that it only irradiates on EMP, it now has use again! I am completely fine with this PR now, and I withdraw my objections.

**Comment by YakumoChen** [comment]
same

**Comment by PKPenguin321** [comment]
Now this PR is pointless nice going

You might as well just remove it again if it's behind EMPs

Science jews get off unnerfed again

**Comment by KorPhaeron** [comment]
Stand strong jordie, radiation should be the price they pay for infinite ammo

**Comment by Jordie0608** [comment]
Is that an invitation to add radiation to the captain's laser? ( ͡° ͜ʖ ͡°)

**Comment by KorPhaeron** [comment]
the price the captain pays for infinite ammo is everyone in the game wanting to grief or kill him

**Comment by GunHog** [comment]
Infinite and self-recharging are not the same thing. "Infinite ammo" means you can fire the weapon continuously without running out. The correct application of this phrase would be to the Kinetic Accelerator, which you can fire continuously without running out of charge.

**Comment by Jordie0608** [comment]
Yes but that's not as funny.

**Pull request merged**